ztp/kube-compare-reference/optional_labels.tmpl

```diff
@@ -0,0 +1,21 @@
+{{- define "allowLabels" }}
+{{- $prefixes := slice . 1 }}
+{{- $result := dict }}
+{{- range $key, $value := (index . 0) }}
+    {{- range $prefix := $prefixes }}
+        {{- if (hasPrefix $prefix $key) }}
+            {{- $_ := set $result $key $value }}
+        {{- end }}
+    {{- end }}
+{{- end }}
+{{- $result | toYaml | nindent 4 }}
+{{- end }}
+
+{{- define "requiredLabels" }}
```

Comment:
I don't see where you're using the first index of the list should it be the starting value of the result?